rustls-mbedcrypto-provider/src/fips_utils.rs

```diff
@@ -0,0 +1,462 @@
+/* Copyright (c) Fortanix, Inc.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! This module contains functions only used with `fips` features
+
+use core::fmt;
+use std::{borrow::Cow, sync::Arc};
+
+use rustls::OtherError;
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum FipsCheckError {
+    Mbedtls(mbedtls::Error),
+    Other(Cow<'static, str>),
+}
+
+impl fmt::Display for FipsCheckError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
```

Comment:
Nit: maybe a nicer impl? This one shows strings as string literals for example.

Reply:
updated